src/helpers.js

```diff
@@ -179,8 +179,34 @@ exports.allowedZero = function(val, char, caretPos, options) {
  * @param {val} string value to convert to a number
  * @param {options} Finput options object
  */
-exports.toNumber = function(val, options) {
-  return val && Number(val.replace(new RegExp(`[${options.thousands}]`, 'g'), ''));
+exports.toNumber = function(stringValue, options) {
+  if (!stringValue) return null;
```

Comment:
`rawToFormatted` and `formattedToRaw` are hopefully more descriptive names. Also, decided to use 'is_js' to to proper input validation - we only want to operate on numbers/strings respectively. There's probably a bigger discussion to be had on internal error handling here, but for now I am ensuring we always return number (Nan) or string ('') from respective functions.